app/controllers/api/v2/registration_controller.rb

```diff
@@ -107,6 +112,14 @@ def prepare_host
 
         @host.save!
       end
+
+      def setup_insights
+        return unless params['setup_insights'].present?
```

Comment:
this is false for the explicit false value, if the default is true and user wants to set it to false, it will never work, we need to explicitly check for `nil?`. `nil`, `true`, `false` are 3 different cases. We should save the parameter only if it's explicitly set.